react/SearchResult.js

```diff
@@ -0,0 +1,267 @@
+    if (variables) {
+      query = variables.query || query
+      map = variables.map || map
+      orderBy = variables.orderBy || orderBy
+    }
+    return getPagesArgs(opt, query, map, orderBy, isSelected, type)
+  }
+
+  renderSearchFilters() {
+    if (!this.props.facetsQuery || !this.props.facetsQuery.facets) return
+
+    const { facetsQuery: { facets }, searchQuery } = this.props
+    const query = searchQuery.variables.query
+    const map = searchQuery.variables.map
+    const selecteds = this.getSelecteds(query, map)
+    const isDisabled = this.countSelecteds(selecteds) === 1
+
+    return FACETS_KEYS.map(key => {
+      if (facets[key]) {
+        if (key === 'SpecificationFilters') {
+          return facets[key].map(filter => {
+            return (
+              <SearchFilter key={filter.name} title={filter.name}
+                options={filter.facets} type={key} selecteds={selecteds[key]}
+                getLinkProps={this.getLinkProps} disabled={isDisabled} />)
+          })
+        } else if (key === 'CategoriesTrees') {
+          let categories = []
+          facets[key].map(filter => {
+            if (selecteds.Departments.indexOf(filter.Name.toUpperCase()) !== -1) {
+              categories = categories.concat(filter.Children)
+            }
+          })
+          return (
+            <SearchFilter key={key} title="Categories" options={categories}
+              type="Departments" selecteds={selecteds.Departments}
+              getLinkProps={this.getLinkProps} disabled={isDisabled} />)
+        }
+        return (
+          <SearchFilter key={key} title={key} options={facets[key]}
+            type={key} getLinkProps={this.getLinkProps}
+            disabled={isDisabled} selecteds={selecteds[key]} />)
+      }
+    })
+  }
+
+  getSelecteds(query, map) {
+    const selecteds = {
+      SpecificationFilters: [],
+      Departments: [],
+      Brands: [],
+      FullText: [],
+    }
+
+    if (!query && !map) return selecteds
+
+    const pathValues = query.split('/')
+    const mapValues = map.split(',')
+    pathValues.map((val, i) => {
+      const valDecoded = decodeURI(val.toUpperCase())
+      if (i > mapValues.length - 1) {
+        selecteds.Departments.push(valDecoded)
+      } else {
+        if (mapValues[i] === 'c') {
+          selecteds.Departments.push(valDecoded)
+        } else if (mapValues[i] === 'b') {
+          selecteds.Brands.push(valDecoded)
+        } else if (mapValues[i].indexOf('specificationFilter') !== -1) {
+          selecteds.SpecificationFilters.push(valDecoded)
+        } else if (mapValues[i] === 'ft') {
+          selecteds.FullText = [valDecoded]
+        }
+      }
+    })
+
+    return selecteds
+  }
+
+  countSelecteds(selecteds) {
+    let count = 0
+    for (const key in selecteds) {
+      if (selecteds[key]) {
+        count += selecteds[key].length
+      }
+    }
+    return count
+  }
+
+  getRecordsFiltered() {
```

Comment:
Try to use ES6 and upgrade your code style. We use arrow functions in this case.